packages/pds/src/db/migrations/20230908T224408678Z-email-tokens.ts

```diff
@@ -12,16 +12,49 @@ export async function up(db: Kysely<unknown>, dialect: Dialect): Promise<void> {
     .addPrimaryKeyConstraint('email_token_pkey', ['purpose', 'did'])
     .execute()
 
+  // for lookups where we don't know the did
+  await db.schema.createIndex('email_token_token_idx').on('token').execute()
```

Comment:
This may need to be unique when paired with `purpose` to ensure two dids don't end-up with the same token.  I believe the `passwordResetToken` column dropped below did have this uniqueness constraint on it for that purpose.  

(Relatedly, I forget if that index/unique-constraint automatically gets dropped when the column is dropped.  If so, I wonder if the `DROP INDEX` is not done `CONCURRENTLY` and can cause table locking.)

Reply:
took care of this in a previous PR with a unique index on `(purpose, token)` 👍